airflow/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -419,8 +419,46 @@ def get_job_awslogs_info(self, job_id: str) -> dict[str, str] | None:
 
         :param job_id: AWS Batch Job ID
         """
-        job_container_desc = self.get_job_description(job_id=job_id).get("container", {})
-        log_configuration = job_container_desc.get("logConfiguration", {})
+        job_desc = self.get_job_description(job_id=job_id)
```

Comment:
@vandonr-amz can you please add this context to the PR description ^^^ It would be great to have future users able to immediately understand what's going on here.

Reply:
ok I see your point, but should there really be one log link per job ?
I'm looking at it, and it seems that in the case of a multinode job, there is multiple `log_configuration` (one per node), but from that log config  we get

- the log group
- the region

I'd imagine that multinode batch jobs would not be multi-region ? So that'd would be a constant across all nodes.
And also, I suppose in an overwhelming majority of the cases, the log group would be the same for all nodes (it would be very weird if it wasn't).

Then we get the stream name from the attempts, but this does not depend on the number of nodes. I imagine in most cases there would be one attempt. If there are more, we make the choice of returning the stream name for the last attempt, which makes sense.

The job runs on many nodes, but the logs all end up in the same log stream.

What we can do is iterate on the log configs to make sure they are all sending logs

- to aws
- in the same region
- in the same group

and log a warning if it's not the case.

Reply:
> I'd imagine that multinode batch jobs would not be multi-region ?

AFAIK, Batch resources are resource specific for any type of jobs
- Compute Environment (ECS or EKS clusters)
- Job Definition
- Job Queues

You could configure logging to another region (Cloudwatch) or supported logger drivers. But it configure during creation (register) Batch Job Definition and it couldn't change by submit job. So it should all store in one destination

> The job runs on many nodes, but the logs all end up in the same log stream.

Nope, each node has own logs within unique log stream

https://user-images.githubusercontent.com/3998685/226111922-79641781-069c-4e84-826e-36a7c53c83ef.mp4

Reply:
wow ok that's hmm... surprising.

Reply:
added a commit to log links to all logs